gnt/gnttab_stubs.c

```diff
@@ -73,16 +73,26 @@ stub_gnttab_unmap (value xgh, value array)
 {
         CAMLparam2 (xgh, array);
         int result;
-
-        caml_enter_blocking_section ();
-        result = xengnttab_unmap (_G (xgh), _M (array)->addr,
-                                  _M (array)->len >> XEN_PAGE_SHIFT);
-        caml_leave_blocking_section ();
-
-        if (result != 0)
-          {
-                  caml_failwith ("Failed to unmap grant");
-          }
+	xengnttab_handle* xgt = _G(xgh);
+	void* start_address = _M(array)->addr;
```

Comment:
Would the code clearer without the alias that we create here for `M(array)->addr`? So not declaring `start_address` and using the pointer instead in all places? Because then the NULL-ing becomes more obvious.

Reply:
`_M` accesses an OCaml object, so we need some kind of alias anyway to only access C objects inside the critical section